integration/test/status/test_result.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package status
+
+import (
+	"fmt"
+	"log"
+	"text/tabwriter"
+)
+
+type TestSuiteResult struct {
+	Name             string
+	TestGroupResults []TestGroupResult
+}
+
+func (r TestSuiteResult) GetStatus() TestStatus {
+	for _, result := range r.TestGroupResults {
+		if result.GetStatus() == FAILED {
+			return FAILED
+		}
+	}
+	return SUCCESSFUL
+}
+
+func (r TestSuiteResult) Print() {
+	log.Printf(">>>>>>>>>>>>>>%v<<<<<<<<<<<<<<", r.Name)
+	log.Printf(">>>>>>>>>>>>>>%v<<<<<<<<<<<<<<", string(r.GetStatus()))
+	for _, result := range r.TestGroupResults {
+		result.Print()
+	}
+	log.Printf(">>>>>>>>>>>>>>><<<<<<<<<<<<<<<")
+}
+
+type TestGroupResult struct {
+	Name        string
+	TestResults []TestResult
+}
```

Comment:
This makes me think that it could be pulled out so that `TestSuiteResult` is just a map.

```go
type TestSuiteResult struct {
    Name string
    TestGroupResults map[string][]TestResults
}
```
I think that doing it this way would break some of the abstractions you have like `GetStatus()`, but this feels too nested.

Reply:
I'd like to keep it this way. Just because it's not a newly defined type doesn't mean it's not nested imo! And also I like knowing that the string is a name for each test result